pkg/api/query.go

```diff
@@ -15,17 +15,17 @@ import (
 	"github.com/timescale/promscale/pkg/promql"
 )
 
-func Query(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable, updateMetrics func(handler, code string, duration float64)) http.Handler {
+func Query(conf *Config, queryEngine *promql.Engine, queryable promql.Queryable, updateMetrics updateMetricCallback) http.Handler {
 	hf := corsWrapper(conf, queryHandler(queryEngine, queryable, updateMetrics))
 	return gziphandler.GzipHandler(hf)
 }
 
-func queryHandler(queryEngine *promql.Engine, queryable promql.Queryable, updateMetrics func(handler, code string, duration float64)) http.HandlerFunc {
+func queryHandler(queryEngine *promql.Engine, queryable promql.Queryable, updateMetrics updateMetricCallback) http.HandlerFunc {
```

Comment:
If you do that, then it might be better to transform [calceled](https://github.com/timescale/promscale/pull/1806/files#diff-6627f4c891d535ea1d74cf2ab4a0e14f89f51f99917e9b51737250903254bbd6R113) and [timeout](https://github.com/timescale/promscale/pull/1806/files#diff-6627f4c891d535ea1d74cf2ab4a0e14f89f51f99917e9b51737250903254bbd6R118) into constants